PatientWebApplication/SearchMicroserviceApi/Service/PrescriptionService.cs

```diff
@@ -0,0 +1,248 @@
+        /// <returns> List of all patient prescriptions. </returns>
+        public List<Prescription> GetPrescriptionsForPatient(int idPatient)
+        {
+            return PrescriptionRepository.GetPrescriptionsForPatient(idPatient);
+        }
+
+        /// <summary> This method is calling searchForComments, searchForUsed, searchForMedicines to get list of filtered <c>Prescription</c> of logged patient. </summary>
+        /// /// <param name="prescriptionSearchDto"><c>prescriptionSearchDto</c> is Data Transfer Object of a <c>Prescription</c> that is beomg used to filter precriptions.
+        /// </param>
+        /// <returns> List of filtered patient prescriptions. </returns>
+        public List<Prescription> SimpleSearchPrescriptions(PrescriptionSearchDto prescriptionSearchDto)
+        {
+            return SearchForDoctor(SearchForMedicines(SearchForUsed(SearchForComments(GetPrescriptionsForPatient(prescriptionSearchDto.PatientId), prescriptionSearchDto), prescriptionSearchDto), prescriptionSearchDto), prescriptionSearchDto);
+
+        }
+
+        /// <summary> This method is calling searchForFirstParameter, searchForOtherParameters to get list of filtered <c>Prescription</c> of logged patient. </summary>
+        /// /// <param name="dto"><c>PrescriptionAdvancedSearchDto</c> is Data Transfer Object of a <c>Prescription</c> that is beomg used to filter precriptions.
+        /// </param>
+        /// <returns> List of filtered patient prescriptions. </returns>
+        public List<Prescription> AdvancedSearchPrescriptions(PrescriptionAdvancedSearchDto dto)
+        {
+            return SearchForOtherParameters(GetPrescriptionsForPatient(dto.PatientId), dto, SearchForFirstParameter(GetPrescriptionsForPatient(dto.PatientId), dto));
+        }
+
+        /// <summary> This method is getting list of filtered <c>Prescription</c> that match list of parameters in <c>PrescriptionAdvnacedSearchDto</c></summary>
+        /// <param name="prescriptions"> List of all <c>Prescription</c> of logged user.
+        /// </param>
+        /// <param name="dto"><c>PrescriptionAdvancedSearchDto</c> is Data Transfer Object of a <c>Prescription</c> that is beomg used to filter precriptions.
+        /// </param>
+        /// <param name="firstPrescriptions"> List of <c>Prescription</c> that contains prescriptions that matches first parameter.
+        /// </param>
+        /// <returns> List of filtered patient prescriptions. </returns>
+        private List<Prescription> SearchForOtherParameters(List<Prescription> prescriptions, PrescriptionAdvancedSearchDto dto, List<Prescription> firstPrescriptions)
+        {
+            for (int i = 0; i < dto.RestRoles.Length; i++)
+            {
+                List<Prescription> othersPrescriptions = SearchForOtherParametersValues(dto.RestRoles[i], dto.Rest[i], prescriptions);
+
+                firstPrescriptions = SearchForLogicOperators(dto.LogicOperators[i], othersPrescriptions, firstPrescriptions);
```

Comment:
Write this in form:
"firstPrescriptions = SearchForLogicOperators(dto.LogicOperators[i], SearchForOtherParametersValues(dto.RestRoles[i], dto.Rest[i], prescriptions), firstPrescriptions);"